File: mainStorage/src/secret.py
from Crypto.Cipher import AES
import base64

# Функция для добавления дополнительных символов в ключ, если его длина меньше требуемой
def pad_key(key):
    while len(key) % 16 != 0:
        key += ' '
    return key

# Функция для шифрования сообщения
def pad_message(message):
    block_size = AES.block_size
    padding_length = block_size - (len(message) % block_size)
    padding = bytes([padding_length]) * padding_length
    return message + padding

def unpad_message(padded_message):
    padding_length = padded_message[-1]
    return padded_message[:-padding_length]

def encrypt(message, secret_key):
    padded_message = pad_message(message.encode())
    padded_key = pad_key(secret_key)
    cipher = AES.new(padded_key.encode(), AES.MODE_ECB)
    encrypted_bytes = cipher.encrypt(padded_message)
    return base64.b64encode(encrypted_bytes).decode()

def decrypt(encrypted_message, secret_key):
    padded_key = pad_key(secret_key)
    cipher = AES.new(padded_key.encode(), AES.MODE_ECB)
    decrypted_bytes = cipher.decrypt(base64.b64decode(encrypted_message))
    return unpad_message(decrypted_bytes).decode()

File: verifier/src/main/java/verifier/Secret.java
package verifier;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;
import java.util.Base64;

public class Secret {
    public static final String secretKey = "mandarinmandarin";
    public static String encrypt(String message) throws Exception {
        SecretKeySpec keySpec = new SecretKeySpec(secretKey.getBytes(), "AES");
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.ENCRYPT_MODE, keySpec);
        byte[] encryptedBytes = cipher.doFinal(message.getBytes());
        return Base64.getEncoder().encodeToString(encryptedBytes);
    }

    public static String decrypt(String encryptedMessage) throws Exception {
        SecretKeySpec keySpec = new SecretKeySpec(secretKey.getBytes(), "AES");
        Cipher cipher = Cipher.getInstance("AES");
        cipher.init(Cipher.DECRYPT_MODE, keySpec);
        byte[] encryptedBytes = Base64.getDecoder().decode(encryptedMessage);
        byte[] decryptedBytes = cipher.doFinal(encryptedBytes);
        return new String(decryptedBytes);
    }
}